### PR TITLE
fix(extensions/fetch): Filter out custom HOST headers

### LINKED ISSUE #11017: Users can set "Host" header for `fetch`
The `host` header should not be able to be set by the user. It should always be populated by `reqwest` / `hyper`.

### PR DESCRIPTION
**Pull request by zlumyo**
Closes #11017. Can't find existing tests for _fetch_ (to add a new one). If it exists somewhere, please tell me.
Manual testing performed with simple scripts:
```typescript
// client.ts
await fetch('http://localhost:3000/test', {headers: {'Host': 'example.com'}});

// server.ts
import { Application, Router } from "https://deno.land/x/oak/mod.ts";

const router = new Router();
router
  .get("/test", (context) => {
    console.log(context.response.headers);
  })

const app = new Application();

app.use(router.routes());
app.use(router.allowedMethods());

await app.listen("127.0.0.1:3000");
```

### CODE REVIEW

**Comment by lucacasonato** [comment]
@zlumyo Tests for `fetch` can be found in `cli/tests/unit/fetch_test.ts`.

**Review by bnoordhuis — APPROVED** [review]

File: cli/tests/unit/fetch_test.ts [hunk 1]
```diff
@@ -1149,3 +1149,20 @@ unitTest({}, function fetchWritableRespProps(): void {
   assertEquals(original.status, new_.status);
   assertEquals(new_.headers.get("x-deno"), "foo");
 });
+
+unitTest(
+  { perms: { net: true } },
+  async function fetchFilterOutCustomHostHeader(): Promise<
+    void
+  > {
+    const client = Deno.createHttpClient({});
```
Comment (lucacasonato) [inline cli/tests/unit/fetch_test.ts:1158]: You can remove this client creation and closing a few lines down. It is unused and not needed.

**Review by lucacasonato — APPROVED** [review]
Thanks! Could you sort out the CLA? It will need to be green to merge this.

File: cli/tests/unit/fetch_test.ts (hunk 1 above)
Comment (zlumyo) [reply cli/tests/unit/fetch_test.ts]: Done!

**Comment by zlumyo** [comment]
I've signed CLA. But there are commits from my old machine with different git-username. Is there any solution for this?

**Comment by lucacasonato** [comment]
@zlumyo You will need to link the git committer email from that commit to GitHub in your account settings. Alternatively you can squash all the commits together into a new commit with a committer email that is linked to your GitHub account.

**Comment by zlumyo** [comment]
Squashed to c3d66ef6c621032c6067bcddf3295748f5b6c5d6

**Pull request merged**